src/pykiso/lib/connectors/cc_pcan_can.py

```diff
@@ -369,3 +384,72 @@ def _cc_receive(
         except Exception:
             log.exception(f"encountered error while receiving message via {self}")
             return {"msg": None}
+
+    def _merge_trc(self) -> None:
+        """Merge multiple trc files in one."""
+
+        if isinstance(self.trace_path, str):
+            self.trace_path = Path(self.trace_path)
+
+        # Get the lastest trace files corresponding to the number of traces created
+        list_of_traces = sorted(self.trace_path.glob("*.trc"), key=os.path.getmtime)[
+            -self.trc_count :
+        ]
+
+        try:
+            if self.trace_name is None:
+                # If a log file is not provided, take the fist trace created as result file
+                result_trace = list_of_traces[0]
+            else:
+                # Else write the first trace content in the log file and then remove it
+                result_trace = str(self.trace_path / self.trace_name)
+                with open(list_of_traces[0], "r") as trc:
+                    merged_data = trc.read()
+                with open(result_trace, "w") as merged_trc:
+                    merged_trc.write(merged_data)
+                os.remove(list_of_traces[0])
+
+            list_of_traces.pop(0)
+            # End of the trace header
+            first_message_line = 33
+
+            # Append all trace files
+            for file in list_of_traces:
+                with open(file, "r") as trc:
+                    data = trc.read().splitlines(True)
+                with open(result_trace, "a") as merged_trc:
+                    merged_trc.writelines(data[first_message_line:])
+                os.remove(file)
+
+            self._fix_message_numbers(result_trace, first_message_line)
+
+        except IndexError:
+            log.internal_warning("No trace to merge")
+
+    def _fix_message_numbers(
```

Comment:
What about the timestamps?

Reply:
Actually it is another ticket, PR coming soon 😉